packages/neon/neon/lib/src/theme/neon.dart

```diff
@@ -55,4 +55,25 @@ class NeonTheme extends ThemeExtension<NeonTheme> {
       dialogTheme: NeonDialogTheme.lerp(dialogTheme, other.dialogTheme, t),
     );
   }
+
+  @override
+  int get hashCode => Object.hashAll([
+        branding,
+        colorScheme,
+        dialogTheme,
+      ]);
+
+  @override
+  bool operator ==(final Object other) {
+    if (identical(this, other)) {
+      return true;
+    }
+    if (other.runtimeType != runtimeType) {
+      return false;
+    }
```

Comment:
Is this even necessary with the `other is NeonTheme`? That is basically the same check.